src/components/AssetsPage/index.jsx

```diff
@@ -10,43 +10,97 @@ import { getAssets } from '../../data/actions/assets';
 import edxBootstrap from '../../SFE.scss';
 import styles from './AssetsPage.scss';
 
-class AssetsPage extends React.Component {
+export class AssetsPage extends React.Component {
   constructor(props) {
     super(props);
+
+    const hasAssets = this.hasAssets(props.assetsList);
+
     this.state = {
       checked: false,
+      hasAssets,
     };
   }
 
   componentDidMount() {
     this.props.getAssets(this.props.request, this.props.courseDetails);
   }
 
-  componentDidUpdate() {
-    this.props.getAssets(this.props.request, this.props.courseDetails);
+  componentWillReceiveProps(nextProps) {
+    if (this.state.hasAssets && this.hasNoAssets(nextProps.assetsList)) {
+      this.setState({
+        hasAssets: false,
+      });
+    } else if (!this.state.hasAssets && this.hasAssets(nextProps.assetsList)) {
+      this.setState({
+        hasAssets: true,
+      });
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (JSON.stringify(prevProps.request) !== JSON.stringify(this.props.request)) {
+      this.props.getAssets(this.props.request, this.props.courseDetails);
+    }
   }
 
+  hasNoAssets = assetsList => (
+    assetsList.length === 0
+  )
+
+  hasAssets = assetsList => (
+    assetsList.length > 0
+  )
+
   handleCheckBoxChange = (checked) => {
     this.setState({
       checked,
     });
   }
 
+  renderAssetsDropZone = () => (
+    <WrappedAssetsDropZone
+      maxFileSizeMB={this.props.uploadSettings.max_file_size_in_mbs}
+    />
+  )
+
+  renderAssetsPage = () => (
+    <React.Fragment>
+      <div className={edxBootstrap.col}>
+        { this.renderAssetsDropZone() }
+        <WrappedAssetsFilters />
+      </div>
+      <div className={edxBootstrap['col-10']}>
+        <WrappedAssetsTable />
+        <WrappedPagination />
+      </div>
+    </React.Fragment>
+  )
+
+  renderNoAssetsPage = () => (
+    <React.Fragment>
+      <div className={edxBootstrap.col}>
+        { this.renderAssetsDropZone() }
+      </div>
+      <div className={edxBootstrap['col-10']}>
+        { !this.state.hasAssets && this.renderNoAssetsBody() }
+      </div>
+    </React.Fragment>
+  )
+
+  renderNoAssetsBody = () => (
+    <React.Fragment>
+      <h3>0 files in your course</h3>
+      <h4>Enhance your course content by uploading files such as images and documents.</h4>
```

Comment:
This display text will need to be extracted for i18n purposes, flagging for myself once this gets combined with https://github.com/edx/studio-frontend/pull/84.

[example of a similar transformation](https://github.com/edx/studio-frontend/pull/84/files#diff-e5d111258070606d64e86dc6dbcd0a42L56)